torch/nn/parallel/distributed.py

```diff
@@ -173,6 +173,28 @@ def backward(ctx, *grad_outputs):
 
         return (None, None, *grad_outputs)
 
+
+class JoinHook(NamedTuple):
```

Comment:
Introducing public APIs requires team-discussion + github RFC post. See this RFC for DDP join: https://github.com/pytorch/pytorch/issues/38174